### PR TITLE
Webasto Next Implementation

### PR DESCRIPTION
**Pull request by tsicking**
Beta version. Not fully tested.

TODO: Session Energy value is during a session 0Wh.

### CODE REVIEW

File: io.openems.edge.evcs.webasto.next/src/io/openems/edge/evcs/webasto/next/EvcsWebastoNextImpl.java [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+		super.deactivate();
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+		// Cannot read the gaps, therefore there are so many tasks
+		var modbusProtocol = new ModbusProtocol(this, //
+				new FC3ReadRegistersTask(1000, Priority.HIGH,
+						m(EvcsWebastoNext.ChannelId.CHARGE_POINT_STATE, new UnsignedWordElement(1000)), //
+						m(ManagedEvcs.ChannelId.CHARGE_STATE, new UnsignedWordElement(1001)), //
```
Comment (sebastianasen) [inline io.openems.edge.evcs.webasto.next/src/io/openems/edge/evcs/webasto/next/EvcsWebastoNextImpl.java:141]: Remove. Channel set automatically by the WriteHandler
Noted in the Channel description

**Review by sebastianasen — CHANGES_REQUESTED** [review]
Thanks for implementing.
Find my comments above.

**Review by sebastianasen — APPROVED** [review]

File: io.openems.edge.evcs.webasto.next/src/io/openems/edge/evcs/webasto/next/EvcsWebastoNextImpl.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+				new FC6WriteRegisterTask(5004, //
+						m(EvcsWebastoNext.ChannelId.CHARGE_CURRENT, new UnsignedWordElement(5004))), //
+				new FC6WriteRegisterTask(5006, //
+						m(EvcsWebastoNext.ChannelId.START_CANCEL_CHARGING_SESSION, new UnsignedWordElement(5006))),
+				new FC3ReadRegistersTask(6000, Priority.LOW, //
+						m(EvcsWebastoNext.ChannelId.LIFE_BIT, new UnsignedWordElement(6000))), //
+				new FC6WriteRegisterTask(6000, //
+						m(EvcsWebastoNext.ChannelId.LIFE_BIT, new UnsignedWordElement(6000))) //
+		);
+		this.addStatusListener();
```
Comment (sebastianasen) [inline io.openems.edge.evcs.webasto.next/src/io/openems/edge/evcs/webasto/next/EvcsWebastoNextImpl.java:209]: Addendum: A listener for the currently used phases is missing, as long as the CurrentL1/L2/L3 values are not part of the nature and the logic is not handled by them

**Review by sebastianasen — CHANGES_REQUESTED** [review]

File: io.openems.edge.evcs.webasto.next/src/io/openems/edge/evcs/webasto/next/EvcsWebastoNextImpl.java (hunk 2 above)
Comment (tsicking) [reply io.openems.edge.evcs.webasto.next/src/io/openems/edge/evcs/webasto/next/EvcsWebastoNextImpl.java:213]: I have added the listener. Thanks for the hint.

**Review by sebastianasen — APPROVED** [review]
Thank you.
I also missed the unit 👍

**Review by sfeilmeier — APPROVED** [review]
Thank you guys! I really appreciate the patience, endurance and professionality that went into this Pull Request.

**Pull request merged**